### ISSUE TITLE
Emoji prefixes are always converted to shortcode

### ISSUE THREAD

**Issue by nevans** [dependabot/dependabot-core#4565]
I primarily view git logs in my terminal, and seeing dependabot's shortlog expansions mixed in with my other emoji-prefixed commits is very annoying!  I've tried to configure the `dependabot.yml` commit messages to use emoji, but they just get converted back to shortcode!

I couldn't find where that conversion is happening in this repository.  Perhaps it's an artifact of octokit or the github API?  When I look at the specs for `Dependabot::PullRequestCreator::PrNamePrefixer`, it _looks_ like it should be using a simple unicode emoji... that's what happens in the spec.  But that's not the behavior I'm experiencing.

In addition to my simple config setting not being honored, I also think the following code snippet is buggy:

https://github.com/dependabot/dependabot-core/blob/7444344c03b86112b4801eec92b0909e05dbfe33/common/lib/dependabot/pull_request_creator/pr_name_prefixer.rb#L253-L261

That's the only logic I could find for automatically determining the commit style... and it completely ignores when commits start with actual unicode emoji!

Thanks! 🙂

**Comment by kingyue737** [comment]
I'm facing the same problem. Hopes it can be fixed

**Comment by roniemartinez** [comment]
I agree this is annoying, especially when you log the history in terminal. I always have to manually replace it before merging.

### PR
<img width="853" alt="image" src="https://github.com/dependabot/dependabot-core/assets/2573537/f674fb5d-4ed2-41a2-b358-dc52a27792c1">


### Terminal
<img width="801" alt="image" src="https://github.com/dependabot/dependabot-core/assets/2573537/6e3f5642-853e-4cae-9f0a-6a1a6bd85242">

**Comment by Stadly** [comment]
Any news on this?

**Comment by deivid-rodriguez** [comment]
No, contributions or just research appreciated though.

**Comment by Stadly** [comment]
This issue is still at large!

**Comment by nevans** [comment]
It's very very annoying, and also quite puzzling.  Please fix it!  🥺🙏🏻

**Comment by avdi** [comment]
Just piping up to agree this is very annoying

**Comment by blue2cat** [comment]
I've done some research on this issue. It looks like [message_builder.rb](https://github.com/dependabot/dependabot-core/blob/c76bfd0f745c702ffecc3e2e602e4d2ffb42db53/common/lib/dependabot/pull_request_creator/message_builder.rb#L341) converts the :arrow_up: and :lock: commit emojis to their shortcode variants (`:arrow_up:` and `:lock:`).

This conversion occurs to all commit messages with :arrow_up: and :lock:, including those with custom values passed via `commit-message.prefix`.  

I found the commit that introduced the split (8be42a66e, May 2018), but it doesn't have any direct rationale for the decision to convert commit message emojis generated by Dependabot to shortcodes.  I can't find any current reasons to avoid unicode emojis in commits, especially since they're already included in PR titles.

--- 
In addition to the Dependabot-generated emojis always landing as shortcodes, the gitmoji-detection code is flawed as @nevans indicates above. It's only looking for shortcode-flavored emoji, not pure unicode. I've opened https://github.com/dependabot/dependabot-core/issues/16190 to track that, since I think it's a separate problem.
